xonsh/built_ins.py

```diff
@@ -438,7 +443,7 @@ def run_subproc(cmds, captured=True):
             stdin = prev_proc.stdout
         try:
             proc = Popen(aliased_cmd, universal_newlines=uninew, env=ENV.detype(),
-                         stdin=stdin, stdout=stdout)
+                         stdin=stdin, stdout=stdout, preexec_fn=os.setsid)
```

Comment:
`preexec_fn` and `os.setsid()` seem to be posix only.  Could we make it so that this does not get passed in unless `os.name == 'posix'`  Maybe a dict of kwargs is the best.